wiki/osu!wiki_contribution_guide/en.md

```diff
@@ -0,0 +1,531 @@
+If you have any adversity in using the CLI, you can find and self-learn the GUI (Graphical User Interface) version of Git or programs like [Git Extensions](https://gitextensions.GitHub.io/ "Git Extensions").
+Some editors (ie. Visual Studio Code) may provide a Git interface.
+
+Please be reminded that local copy and remote copy are two different things.
+If you did commits without pushing to remote copy, you are effectively talking to yourself.
+You can delay the push to remote copy for as long as you want, but you must push your commit at the end.
+It is usually preferred to push your commit immediately once you complete a commit.
+
+#### One-time setup only
+
+_Note: If you don't have Git installed, you can go to [their website](https://git-scm.com/) to download Git and [set it up](https://help.GitHub.com/articles/set-up-git/) (only ignore step 1)._
+
+Once you have Git installed, open up Git and enter this in your Git's Command Line Interface (CLI):
+
+```prolog
+git clone https://GitHub.com/<your GitHub username (ie. peppy)>/osu-wiki
+```
+
+This will clone your fork in the filepath you specified (usually in your root folder of the drive Git was installed, when running Git directly).
+
+Now to access it...
+
+_Note: For further reference, check GitHub's Help on fork a repo: <https://help.GitHub.com/articles/fork-a-repo/>_
+
+#### Accessing your local copy
+
+Open up Git (if you have not) and type in this command in Git's CLI:
+
+```prolog
+git status
+```
+
+This command is used to check for Git's response about the status of the local copy in this directory.
+
+If it throws up an error (which is an expected outcome in this situation), type in this command in Git's CLI:
+
+```prolog
+cd osu-wiki
+```
+
+Assuming you are following what was written (meaning, you are in root folder, `~`) above, the terminal command, `cd`, will move Git into the proper filepath.
+There should be a light-blue coloured brackets with text inside at the end of the line.
+That is the branch name your local copy is currently in.
+
+You can now proceed to make your changes.
+
+#### Branching
+
+To check for available branches, enter the command below in Git's CLI:
+
+```prolog
+git branch
+```
+
+This command will show all the branches of the remote repository (namely, your forked repo, nicknamed ``origin`` by default) copied into your local copy.
+If you are starting fresh, you will be in ``(master)`` branch by default.
+
+To switch to a branch, enter the command below in Git's CLI:
+
+```prolog
+git checkout <branchname>
+```
+
+Once completed, your local copy will be updated exactly to the branch copy version you switched to.
+You are now in the branch specified in ``<branchname>``.
+If you forgot to save the files before switching, try and see if your text editor still keeps it or not.
+
+To make a new branch, enter the commands below in Git's CLI:
+
+```prolog
+git checkout master
+git checkout -b <branchname>
+git push origin <branchname>
```

Comment:
Locally, I have made a new branch (yay).
In my remote repo? No (because no connection situation, and _nobody knows_).

The push is to tell the GitHub to make the branch for me based on the local branch version (once I get my connection up and alive).

It need a little bit of **aha!** situation to get the reason why.